services/controller/src/plz/controller/images/images_base.py

```diff
@@ -15,8 +15,11 @@
 
 
 class Images(ABC):
-    def __init__(self, docker_api_client: docker.APIClient, repository: str):
-        self.docker_api_client = docker_api_client
+    def __init__(self,
+                 docker_api_client_creator: Callable[[], docker.APIClient],
+                 repository: str):
+        self.docker_api_client_creator = docker_api_client_creator
```

Comment:
Why store this?

Reply:
The whole point is that you store the creator and the client. You use the creator to create a new client in each login

Reply:
(But reuse the client otherwise)